### PR TITLE
Fix collision exception when the scale is negative

### PR DESCRIPTION
**Pull request by luzhuang**
link: https://github.com/galacean/runtime/issues/1826

### CODE REVIEW

File: packages/physics-lite/src/LitePhysicsScene.ts [hunk 1]
```diff
@@ -243,7 +245,9 @@ export class LitePhysicsScene implements IPhysicsScene {
       } else if (myShape instanceof LiteSphereColliderShape) {
         LitePhysicsScene._upWorldSphere(myShape, this._sphere);
         for (let j = 0, len = colliders.length; j < len; j++) {
-          const colliderShape = colliders[j]._shapes;
+          const otherCollider = colliders[j];
+
+          const colliderShape = otherCollider._shapes;
```
Comment (GuoLei1990) [inline packages/physics-lite/src/LitePhysicsScene.ts:250]: revert this

File: packages/physics-lite/src/LitePhysicsScene.ts [hunk 2]
```diff
@@ -216,7 +216,9 @@ export class LitePhysicsScene implements IPhysicsScene {
       if (myShape instanceof LiteBoxColliderShape) {
         LitePhysicsScene._updateWorldBox(myShape, this._box);
         for (let j = 0, len = colliders.length; j < len; j++) {
-          const colliderShape = colliders[j]._shapes;
+          const otherCollider = colliders[j];
+
+          const colliderShape = otherCollider._shapes;
```
Comment (GuoLei1990) [inline packages/physics-lite/src/LitePhysicsScene.ts:221]: revert this

File: packages/physics-physx/src/shape/PhysXBoxColliderShape.ts [hunk 3]
```diff
@@ -11,17 +11,13 @@ export class PhysXBoxColliderShape extends PhysXColliderShape implements IBoxCol
...
     super(physXPhysics);
 
     this._halfSize.set(size.x * 0.5, size.y * 0.5, size.z * 0.5);
-
-    this._pxGeometry = new physXPhysics._physX.PxBoxGeometry(
-      this._halfSize.x * this._worldScale.x,
-      this._halfSize.y * this._worldScale.y,
-      this._halfSize.z * this._worldScale.z
-    );
+    this._pxGeometry = new physXPhysics._physX.PxBoxGeometry(this._halfSize.x, this._halfSize.y, this._halfSize.z);
```
Comment (GuoLei1990) [inline packages/physics-physx/src/shape/PhysXBoxColliderShape.ts:20]: Cache  `halfSize`

File: packages/physics-physx/src/shape/PhysXBoxColliderShape.ts [hunk 4]
```diff
@@ -44,9 +40,13 @@ export class PhysXBoxColliderShape extends PhysXColliderShape implements IBoxCol
    */
   override setWorldScale(scale: Vector3): void {
     super.setWorldScale(scale);
-
+    this._sizeScale.copyFrom({
+      x: Math.abs(scale.x),
+      y: Math.abs(scale.y),
+      z: Math.abs(scale.z)
+    });
```
Comment (GuoLei1990) [inline packages/physics-physx/src/shape/PhysXBoxColliderShape.ts:47]: Why new Object?

File: packages/physics-lite/src/LitePhysicsScene.ts (hunk 2 above)
Comment (luzhuang) [reply packages/physics-lite/src/LitePhysicsScene.ts]: done

File: packages/physics-physx/src/shape/PhysXBoxColliderShape.ts (hunk 3 above)
Comment (luzhuang) [reply packages/physics-physx/src/shape/PhysXBoxColliderShape.ts]: done

File: packages/physics-physx/src/shape/PhysXBoxColliderShape.ts (hunk 4 above)
Comment (luzhuang) [reply packages/physics-physx/src/shape/PhysXBoxColliderShape.ts]: done

**Pull request merged**